hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -1103,10 +1104,15 @@ public void memberAdded(MembershipEvent membershipEvent) {
 
             @Override
             public void memberRemoved(MembershipEvent membershipEvent) {
-                if (isMaster()) {
+                if (isMaster(membershipEvent) && jobRepository.jobRecordsExists()) {
```

Comment:
If the local member isn't a master, it won't initialize `JobRepository.jobRecords`. Now, when a master is removed and this member becomes a master, what would initialize the local handle? 
I suspect you wanted to test the existence of the map, but this code tests the existence of the handle to the map.